Google Search Console/v1 (#544)
stitchdata/docs: _saas-integrations/google-search-console/v1/google-search-console-v1.md

loading-reports: true

table-selection: true
column-selection: true

# attribution-window: "# days"
# attribution-is-configurable: 

# setup-name: ""

# -------------------------- #
#      Feature Summary       #
# -------------------------- #

feature-summary: |
  Stitch's {{ integration.display_name }} integration replicates data using the {{ integration.api | flatify | strip }}. Refer to the [Schema](#schema) section for a list of objects available for replication.

# -------------------------- #
#      Incompatibilities     #
# -------------------------- #

## uncomment section below if integration is compatible with any Stitch destinations
## if incompatible with multiple destinations, create a section for each destination

## incompatible:
  ## [redshift]: "always,sometimes,never"
  ## reason: "copy" 

# -------------------------- #
#      Setup Instructions    #
# -------------------------- #

requirements-list:
  - item: |
      **An admin account**. Admin access is required in your {{ integration.display_name }} account to change certain settings.

setup-steps:
  - title: "Set up your {{ integration.display_name }}"
    anchor: "set-up-console"
    content: |

>       If you don't have any URLs set up in your {{ integration.display_name }} account, use the following instructions. If you do, you can skip to [step 2](#create-project).

Reviewer: I'd use a callout to make this info stand out.

      1. Login to your Google account and navigate to [{{ integration.display_name }}](https://search.google.com/search-console).
      2. Add the domains or websites for your organization. If adding a domain, it will cover all URLs across all of its subdomains. If adding a website, only the entered websites will be counted for.
      3. Verify your site ownership by clicking **CONTINUE**. Depending on your verification method, this can take up to a day.
      4. Once your link(s) are verified, add them to the `sitemap.xml` files in the **Sitemaps** section of your {{ integration.display_name }} account. Click **Submit**.

  - title: "Create a {{ integration.display_name }} project"
    anchor: "create-project"
    content: |

>       You can skip to [step 3](#enable-apis) if you already have a project in your {{ integration.display_name }} account. If not, use the following instructions.

Reviewer: I'd use a callout to make this info stand out.

      1. Login to your [**Google Cloud Console**](https://console.cloud.google.com).
      2. Click **Create New Project**.
      3. Enter a **Project name** and select a project **Location**.
      4. Click **CREATE**.

  - title: "Enable {{ integration.display_name }} API"
    anchor: "enable-apis"
    content: |
      1. Login to your **Google Cloud Platform**. Make sure you are inside the project that you will use for your {{ integration.display_name }} integration.
      2. Select **APIs & Services** from the left-hand menu panel. This will take you to the **APIs & Services** dashboard.
      3. In the dashboard, click **ENABLE APIS AND SERVICES**. You will be redirected to the **API Library**.
      4. Search for {{ integration.display_name }} and select `{{ integration.display_name }} API`.
      5. Click **ENABLE**.

  - title: "Verify API domain"
    anchor: "verify-api-domain"
    content: |    
      1. Staying in the **APIs & Services** section, click on **Domain verification** in the left-hand menu panel.
      2. Click **Add domain**.
      3. Add the domain you're using for this project.

  - title: "API OAuth consent"
    anchor: "oauth-consent"
    content: |

>       1. Staying in the **APIs & Services** section, click on **OAuth consent screen** in the left-hand menu panel.

Reviewer: ```suggestion
      1. Staying in the **APIs & Services** section, click on the **OAuth consent screen** in the left-hand menu panel.
```

      2. Select your project.
      3. Make the **Application type** `Public`.
      4. Make the **Application name** `tap-google-search-console`.

>       5. Add an image for a logo to help you identify your app.

Reviewer: Is this optional?

      6. In the **Support email** field, enter your email that you will use for this app.
      7. In the **Scope** section, click **Add scope**.
      8. Add the following scope: `Google Search Console API ../auth/webmasters.readonly`.

>       9. In the **Authorized domains** section, add your domains that you added from [step 1](#set-up-console).

Reviewer: ```suggestion
      9. In the **Authorized domains** section, add the domains that you added to {{ integration.display_name}} in [step 1](#set-up-console).
```

      10. Click **SAVE**.

>       11. You will be taken to a screen that says **OAuth grant limits**. The default raise limit is one hour. You may increase the raise limit up to 30 days.

Reviewer: `raise` limit? Is this intentional?

  - title: "Create OAuth credentials"
    anchor: "create-credentials"
    content: |
      1. Staying in the **APIs & Services** section, click on **Credentials** in the left-hand menu panel.
      2. Click on the **Create credentials** dropdown menu.
      3. Select **OAuth client ID**.
      4. In the **Create OAuth client ID** page, select **Web application** for the Application type.
      5. In the **Name** field, enter `tap-google-search-console`.

>       6. In the **Authorized redirect URIs** section, add your domains that you added from [step 1](#set-up-console).

Reviewer: ```suggestion
      6. In the **Authorized redirect URIs** section, add the domains that you added in [step 1](#set-up-console).
```

      9. In the **Authorized domains** section, add the domains that you added to {{ integration.display_name}} in [step 1](#set-up-console).
      10. Click **SAVE**.
      11. You will be taken to a screen that says **OAuth grant limits**. The default limit is 100 grants per minute, per day. You can request to raise the grant limit for your project by clicking **Raise limit**. You can also change the time interval grant resets. The options for time intervals are: `1h`, `6h`, `1d,`, `7d`, and `30d`.

  - title: "Create OAuth credentials"
    anchor: "create-credentials"
    content: |
      1. Staying in the **APIs & Services** section, click on **Credentials** in the left-hand menu panel.
      2. Click on the **Create credentials** dropdown menu.
      3. Select **OAuth client ID**.
      4. In the **Create OAuth client ID** page, select **Web application** for the Application type.
      5. In the **Name** field, enter `tap-google-search-console`.
      6. In the **Authorized redirect URIs** section, add the domains that you added in [step 1](#set-up-console).
      7. Click **CREATE**.

  - title: "add integration"
    content: |

>       4. In the **Site Urls** field, enter the URLs you added to your sitemap file in [step 1](#set-up-console). Make sure your URLs begin with `https://` or `http://`.

Reviewer: This is close to what I meant, but not entirely what I was getting at. *How* should the URLs be listed in this field? Are they separated by commas, pipes, something else? What does the tap's README say?

Also - whatever the answer, it'll need to be updated here as well as the API object.
